### DISCUSSION TITLE
Connect ESP32 to remote ESP8266

### DISCUSSION THREAD

**Discussion by achati04** [HomeSpan/HomeSpan#605] [Q&A] [answered]
Hello everyone,

I would like to start using HomeSpan to setup my own smart home devices, I am new to ESPs and Arduino but I am starting to understand how to setup everything wired, using the project doc. But now, I am looking further to make things work wirelessly, instead of connecting my LEDs, Switches and other devices to my ESP32, I would like to connect each device to a separate ESP8266, much cheaper and smaller, that I want to connect wirelessly to my main ESP32, that will centralize everything.  Basically, I want to have many ESP8266, each one is going to control one device, and every ESP8266 will be connected wirelessly to the main ESP32. The thing is that idk how to do it ... If anyone could explain how I should proceed or even give me a basic code example that I could rely on. 

Thanks for your help !

**Comment by HomeSpan** [comment, marked as answer]
You can accomplish this with ESP-NOW using HomeSpan's integrate [SpanPoint Class](https://github.com/HomeSpan/HomeSpan/blob/master/docs/NOW.md).  Note this is a fairly advanced use case.  I recommend you first make sure you are very comfortable with using HomeSpan locally, wiring LEDs, switches, etc., pairing/unpairing, until everything works perfectly, and then extend the project to wireless communication with an ESP8266 using SpanPoint.
